cockatrice/src/SpoilerBackgroundUpdaterThread.cpp

```diff
@@ -0,0 +1,233 @@
+#include <QDateTime>
+#include <QDebug>
+#include <QUrl>
+#include <QNetworkReply>
+#include <QMessageBox>
+#include <QFile>
+#include <QApplication>
+#include <QDesktopWidget>
+#include <QtConcurrent>
+#include "SpoilerBackgroundUpdaterThread.h"
+#include "settingscache.h"
+#include "carddatabase.h"
+#include "main.h"
+
+SpoilerBackgroundUpdaterThread::SpoilerBackgroundUpdaterThread(QObject *apParent) : QObject(apParent)
+{
+    // Determine if we're in spoilers season or not. If we're not, just kill this thread
+    downloadSpoilerSeasonStatusFile();
```

Comment:
This really needs to be _after_ the settings check. If someone doesn't want spoilers then there's an unnecessary request being made here.